homework/index.js

```diff
@@ -30,18 +32,108 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+  function createDescription(selectedRepository, container) {
+    const descriptionContainer = createAndAppend('div', container, {
+      class: 'left-div',
+    });
+    const table = createAndAppend('table', descriptionContainer, { class: 'table' });
+    const tbody = createAndAppend('tbody', table);
+    const detailsHeader = [{ title: 'Repository', value: selectedRepository.name }];
+    const detailsBody = [
+      { title: 'Description', value: selectedRepository.description },
+      { title: 'Forks', value: selectedRepository.forks },
+      { title: 'Updated', value: new Date(selectedRepository.updated_at).toLocaleString() },
+    ];
+    detailsHeader.forEach(detail => {
```

Comment:
1. Only the first row needs special treatment because of the `<a>` tag. Therefore, not need to use an array with `.forEach()` here.
2. Your code becomes easier to understand if you place related code together and potentially separate the code sections with a blank line:

```js
function createDescription(selectedRepository, container) {
  const descriptionContainer = createAndAppend('div', container, {
    class: 'left-div',
  });
  const table = createAndAppend('table', descriptionContainer, { class: 'table' });
  const tbody = createAndAppend('tbody', table);

  const detailsHeader = { title: 'Repository', value: selectedRepository.name };
  const trRepository = createAndAppend('tr', tbody);
  createAndAppend('td', trRepository, { text: detailsHeader.title, class: 'label' });
  createAndAppend('a', trRepository, {
    text: detailsHeader.value,
    href: selectedRepository.html_url,
    target: '_blank',
  });

  const detailsBody = [
    { title: 'Description', value: selectedRepository.description },
    { title: 'Forks', value: selectedRepository.forks },
    { title: 'Updated', value: new Date(selectedRepository.updated_at).toLocaleString() },
  ];
  detailsBody.forEach(detail => {
    const tr = createAndAppend('tr', tbody);
    createAndAppend('td', tr, { text: detail.title, class: 'label' });
    createAndAppend('td', tr, { text: detail.value, class: 'repository-data' });
  });
}
```